From 6de324a5c613664b7ec3162fc399c1fc4f4b7665 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Thu, 30 Jan 2025 17:06:12 -0800
Subject: [PATCH 13/13] fix the build on LMDE 6

GCC 12.20 on LDME6 has some bug or is even more strict
for handling the warnings as an error than the gcc13/14
on fedora. I had earlier the

CMAKE_CXX_FLAGS -Wno-error=maybe-uninitialized
but for LDME6, i need to add another flag also:
$CMAKE_CXX_FLAGS -Wno-error=maybe-uninitialized -Wno-error=uninitialized

I think these warnings are fixed in newer fbgemm which is one
sub-project that pytorch uses. For stable pytorch version the easiest way to fix
this now is just disable gcc warnings to be triggered as an error.

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 build_rocm.sh | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/build_rocm.sh b/build_rocm.sh
index 86ac0cb005e..77c69772565 100755
--- a/build_rocm.sh
+++ b/build_rocm.sh
@@ -23,9 +23,11 @@ else
 	echo "Building Pytorch for GPU's used on rocm sdk build."
 fi
 unset LDFLAGS
-export CFLAGS="-Wno-error=maybe-uninitialized"
+#export CFLAGS="-Wno-error=maybe-uninitialized -Wno-error=uninitialized"
+#export CPPFLAGS="-Wno-error=maybe-uninitialized -Wno-error=uninitialized"
+unset CFLAGS
 unset CPPFLAGS
 unset PKG_CONFIG_PATH
 export UCC_HOME=$install_dir_prefix_rocm
 export UCX_HOME=$install_dir_prefix_rocm
-USE_SYSTEM_UCC=1 USE_UCC=1 BUILD_TEST=0 USE_FLASH_ATTENTION=ON AOTRITON_INSTALLED_PREFIX=${install_dir_prefix_rocm} ROCM_PATH=${install_dir_prefix_rocm} ROCM_SOURCE_DIR=${install_dir_prefix_rocm} CMAKE_CXX_FLAGS="$CMAKE_CXX_FLAGS -Wno-error=maybe-uninitialized" CMAKE_PREFIX_PATH="${install_dir_prefix_rocm};${install_dir_prefix_rocm}/lib64/cmake;${install_dir_prefix_rocm}/lib/cmake;${install_dir_prefix_rocm}/lib64;${install_dir_prefix_rocm}/lib" ROCM_VERSION=${rocm_version_str} HIP_ROOT_DIR=${install_dir_prefix_rocm} USE_ROCM=1 PYTORCH_BUILD_VERSION="$(git describe --tags --abbrev=0 | sed 's/^v//')" PYTORCH_BUILD_NUMBER=1 python setup.py bdist_wheel
+USE_SYSTEM_UCC=1 USE_UCC=1 BUILD_TEST=0 USE_FLASH_ATTENTION=ON AOTRITON_INSTALLED_PREFIX=${install_dir_prefix_rocm} ROCM_PATH=${install_dir_prefix_rocm} ROCM_SOURCE_DIR=${install_dir_prefix_rocm} CMAKE_C_FLAGS="$CMAKE_C_FLAGS -Wno-error=maybe-uninitialized -Wno-error=uninitialized" CMAKE_CXX_FLAGS="$CMAKE_CXX_FLAGS -Wno-error=maybe-uninitialized -Wno-error=uninitialized" CMAKE_PREFIX_PATH="${install_dir_prefix_rocm};${install_dir_prefix_rocm}/lib64/cmake;${install_dir_prefix_rocm}/lib/cmake;${install_dir_prefix_rocm}/lib64;${install_dir_prefix_rocm}/lib" ROCM_VERSION=${rocm_version_str} HIP_ROOT_DIR=${install_dir_prefix_rocm} USE_ROCM=1 PYTORCH_BUILD_VERSION="$(git describe --tags --abbrev=0 | sed 's/^v//')" PYTORCH_BUILD_NUMBER=1 python setup.py bdist_wheel
-- 
2.39.5

